network/src/transport.rs

```diff
@@ -301,21 +319,33 @@ where
         chain_id: ChainId,
         application_protocols: SupportedProtocols,
     ) -> Self {
-        let mut own_handshake = HandshakeMsg::new(chain_id, network_context.network_id().clone());
-        own_handshake.add(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
+        // build supported protocols
+        let mut supported_protocols = BTreeMap::new();
+        supported_protocols.insert(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
+
+        // get network public key
         let identity_pubkey = identity_key.public_key();
 
+        // auth mode
+        // TODO(mimoo): should we build this based on the networkid, and not on trusted peers
         let auth_mode = match trusted_peers.as_ref() {
             Some(trusted_peers) => HandshakeAuthMode::mutual(trusted_peers.clone()),
             None => HandshakeAuthMode::ServerOnly,
         };
 
+        // create upgrade context
+        let network_id = network_context.network_id().clone();
+        let upgrade_context = UpgradeContext {
+            noise: NoiseUpgrader::new(network_context, identity_key, auth_mode),
+            handshake_version,
+            supported_protocols,
+            chain_id,
+            network_id,
+        };
+
+        //
```

Comment:
??

Reply:
(separation of return value)

Reply:
??  No empty comments please.  Whitespace can do the separation itself.

Reply:
Sure, that's my style but I can see how people would not like it